.github/workflows/ci.yml

```diff
@@ -189,6 +189,8 @@ jobs:
           VER: '5.5.2'
         - SWIGLANG: scilab
           os: ubuntu-18.04 # scilab 6.0
+        - SWIGLANG: scilab
+          os: ubuntu-22.04 # scilab 6.1
```

Comment:
22.04 actually has scilab 6.1.1 (from the CI output and also from looking at the package repos, see output below) while the existing job below this addition is run on ubuntu-20.04 which has scilab 6.1.0.

It doesn't seem useful to test both 6.1.0 and 6.1.1, or at least not useful enough when we already have so many CI jobs that the time and resources to run them all are becoming concerning.  If the "0.0.1" difference is actually more significant here than I'm naively expecting, please speak up.

I've push a change to drop the 6.1.0 job, resurrect the 6.0 job (github stopped supporting their ubuntu-18.04 image recently so all the 18.04 jobs started to fail and I had to disable them) and add a 2023.0.0 build, since there seems to be a binary build on scilab.org for it and really we should have a job to confirm this new support works.

---

Scilab versions packaged for Ubuntu are:

```
$ grep 'LTS' /usr/share/distro-info/ubuntu.csv
6.06 LTS,Dapper Drake,dapper,2005-10-12,2006-06-01,2009-07-14,2011-06-01
8.04 LTS,Hardy Heron,hardy,2007-10-18,2008-04-24,2011-05-12,2013-05-09
10.04 LTS,Lucid Lynx,lucid,2009-10-29,2010-04-29,2013-05-09,2015-04-30
12.04 LTS,Precise Pangolin,precise,2011-10-13,2012-04-26,2017-04-28,2017-04-28,2019-04-28
14.04 LTS,Trusty Tahr,trusty,2013-10-17,2014-04-17,2019-04-25,2019-04-25,2024-04-25
16.04 LTS,Xenial Xerus,xenial,2015-10-22,2016-04-21,2021-04-21,2021-04-21,2026-04-23
18.04 LTS,Bionic Beaver,bionic,2017-10-19,2018-04-26,2023-05-31,2023-05-31,2028-04-26
20.04 LTS,Focal Fossa,focal,2019-10-17,2020-04-23,2025-05-29,2025-05-29,2030-04-23
22.04 LTS,Jammy Jellyfish,jammy,2021-10-14,2022-04-21,2027-06-01,2027-06-01,2032-04-21
$ rmadison -uubuntu scilab
 scilab | 5.5.0-2                | trusty/universe          | source, all
 scilab | 5.5.2-2ubuntu3         | xenial/universe          | source, all
 scilab | 6.0.1-1ubuntu1         | bionic/universe          | source, all
 scilab | 6.0.1-7ubuntu1~18.04   | bionic-security/universe | source, all
 scilab | 6.0.1-7ubuntu1~18.04.2 | bionic-updates/universe  | source, all
 scilab | 6.1.0+dfsg1-1ubuntu3   | focal/universe           | source, all
 scilab | 6.1.0+dfsg1-1ubuntu3.1 | focal-updates/universe   | source, all
 scilab | 6.1.1+dfsg2-3ubuntu1   | jammy/universe           | source, all
 scilab | 6.1.1+dfsg2-4          | kinetic/universe         | source, all
 scilab | 6.1.1+dfsg2-5          | lunar/universe           | source, all
```

Reply:
The 2023.0.0 binary dowload currently fails with HTTP error 403 (Forbidden).  I've reported this to scilab:

https://gitlab.com/scilab/scilab/-/issues/17024

Reply:
I've merged my change to resurrect the scilab 6.0 CI job and the fix for #894 (which really shouldn't be mixed up with these unrelated changes).

I'm a bit hesitant to merge the 2023.0.0 support without the CI job working since it seems likely the 403 issue will get fixed pretty quickly, so let's give them a day or to and see if it's working.  Don't worry about rebuilding the branch as I have it pretty much ready to go locally.

Reply:
actually the scilab binary file as a xz file not a gz file. As this can be good to distinguish this CI download, I think we can also ship a gz file for SWIG.